src/odemis/driver/pwrmccdaq.py

```diff
@@ -0,0 +1,420 @@
+        self._swVersion = odemis.__version__
+        self._metadata[model.MD_SW_VERSION] = self._swVersion
+        self._metadata[model.MD_HW_VERSION] = self._hwVersion
+
+    def _power_to_volt(self, power: float, curve: List[Tuple[float, float]]) -> float:
+        """
+        Calculate the power to the right voltage using the specified power curve
+        :param power (0 < float): the requested power of the light
+        :param curve (list of tuple (float, float)): the mapping between volt -> power
+        :return (float): voltage for outputting the given power
+        raise (ValueError): if the requested power value is out of the power curve range limits
+        """
+        if power < curve[0][1]:
+            raise ValueError("Power requested %g < %g" % (power, curve[0][1]))
+
+        # Find the segment that correspond to that power
+        for i, (v, w) in enumerate(curve[1:]):
+            if power <= w:
+                seg = i
+                break
+        else:
+            raise ValueError("Power requested %g > %g" % (power, curve[-1][1]))
+
+        logging.debug("Converting %g W using segment %d: %s -> %s",
+                      power, seg, curve[seg], curve[seg + 1])
+
+        basev, basew = curve[seg]
+        endv, endw = curve[seg + 1]
+
+        ratio = (power - basew) / (endw - basew)
+        v = basev + ratio * (endv - basev)
+        return v
+
+    def _update_power(self, value: List[float]):
+        for ao_ch, do_ch, crv, pwr in zip(self._ao_channels, self._do_channels, self._pwr_curve, value):
+            pwr = min(pwr, crv[-1][1])
+            volt = self._power_to_volt(pwr, crv)
+            data = int((volt / MAX_VOLTAGE) * MAX_VOLTAGE_VALUE)  # data input expects an uint16
+            # update the analogue output value
+            logging.debug(f"Setting ao_channel {ao_ch} to {volt} V = {pwr} W")
+            self.device.AOut(ao_ch, data)
+
+            port, bit = MCCDevice.channel_to_port(do_ch)
+            old_bit_value = self.device.DBitIn(port, bit)
+            new_bit_value = int(pwr > 0.0)
+            # update the digital output value by using a direct digital port bit
+            if old_bit_value != new_bit_value:
+                logging.debug(f"Setting do_channel {do_ch} from {old_bit_value} to {new_bit_value}")
+                self.device.DBitOut(port, bit, new_bit_value)
+
+
+class MCCDeviceSimulator(object):
+    """
+    A really basic and simple interface to simulate a USB-1208LS device with
+    support for DIO pin read/write commands and a single AO write command.
+    """
+
+    def __init__(self):
+        # initialize values
+        self.productID = 0x0007a  # USB-1208LS
+        # to keep track of the individual bits and values
+        self.port_a_bit_config = {"0": 0, "1": 0, "2": 0, "3": 0, "4": 0, "5": 0, "6": 0, "7": 0}
+        self.port_b_bit_config = {"0": 1, "1": 1, "2": 1, "3": 1, "4": 1, "5": 1, "6": 1, "7": 1}
```

Comment:
Instead of a dict, just use a list of 8 integers (or bool). They are already accessible as config[0] to config[7].

Alternatively, you could just store a single int, and do the bit arithmetic in the function. ;-)